framework/include/geomsearch/NearestNodeLocator.h

```diff
@@ -80,6 +81,100 @@ class NearestNodeLocator : public Restartable
     Real _distance;
   };
 
+  template <unsigned int KDDim> class PointListAdaptor {
+  private:
+    const std::vector<Point> &_pts;
+
+  public:
+    PointListAdaptor(const std::vector<Point> &pts) : _pts(pts) {}
+
+    /**
+     * libMesh \p Point coordinate type
+     */
+    typedef Real coord_t;
+
+    /**
+     * Must return the number of data points
+     */
+    inline size_t kdtree_get_point_count() const { return _pts.size(); }
```

Comment:
I had the same comment, but I think that nanoflann expects this method in the templated class that users are supposed to write. Note that she does not call it anywhere, so I think it gets called from inside the nanoflann magic.